library/ssl_tls.c

```diff
@@ -1566,6 +1566,8 @@ static int ssl_encrypt_buf( mbedtls_ssl_context *ssl )
 #if defined(MBEDTLS_SSL_ENCRYPT_THEN_MAC)
         if( auth_done == 0 )
         {
+            unsigned char mac[MBEDTLS_SSL_MAC_ADD];
```

Comment:
The point is that `mbedtls_md_hmac_finish` might write more than `maclen` bytes if the truncated HMAC extension is used (in which case `maclen == 10`), and in this case, we would overflow the record buffer if we passed `ssl->out_iv + ssl->out_msglen` directly.

Reply:
Ok, I will enhance the commit message. 

The overflow is non-critical because for CBC we always account for up to 256 bytes of CBC padding in the computation of `MBEDTLS_SSL_OUT_BUFFER_LEN` from `MBEDTLS_SSL_OUT_CONTENT_LEN`, but we actually use minimal padding during encryption, so the MAC computation will not go beyond the bounds of the record heap buffer.